frontend/app/src/store/defi/actions.ts

```diff
@@ -265,39 +265,62 @@ export const actions: ActionTree<DefiState, RotkehlchenState> = {
     }
   },
 
-  async fetchAllDefi({ commit, dispatch, state }, refreshing: boolean = false) {
+  async fetchAllDefi(
+    { commit, dispatch, state, rootState: { session } },
+    refreshing: boolean = false
+  ) {
     if (
       state.status === Status.LOADING ||
       (state.status === Status.LOADED && !refreshing)
     ) {
       return;
     }
 
+    const { activeModules } = session!.generalSettings;
+    const noActiveModules = activeModules.length === 0;
+
     commit('status', refreshing ? Status.REFRESHING : Status.LOADING);
     commit('defiStatus', refreshing ? Status.REFRESHING : Status.LOADING);
     await dispatch('fetchDefiBalances');
     commit('defiStatus', Status.LOADED);
-    await dispatch('fetchAaveBalances');
-    await dispatch('fetchDSRBalances');
-    await dispatch('fetchMakerDAOVaults');
+
+    if (noActiveModules || activeModules.includes('aave')) {
```

Comment:
Kelsos didn't we say that if there are no active modules it means none of them are queried? Comment is [here](https://github.com/rotki/rotki/pull/1272#pullrequestreview-462064587) and is in bold.

We need to have a way for a user to disable all module if he is not interested in any of them.

Reply:
Yes completely missed/forgot about it give me a moment I am fixing it